### PR TITLE
aiida-core 2.x compatibility changes

### LINKED ISSUE #40: Preparation for aiida 2.x compatibility
In the develop branch most of the things should be compatible with `aiida-2.x`. But it needs to be thoroughly tested and make sure that no outstanding deprecation warnings are left. The idea of this issue is to act as a placeholder to ensure that we can discuss if anything needs to be done for this.

### PR DESCRIPTION
**Pull request by JPchico**
Changing the code so that it is compatible with aiida-2.x. Fixes #40

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ classifiers = [
 keywords = ["aiida", "workflows", "lammps"]
 requires-python = ">=3.8"
 dependencies = [
-    "aiida-core[atomic-tools]>=1.4.0,<2.0.0",
+    "aiida-core[atomic-tools]>=2.0.0,<3.0.0",
```
Comment (chrisjsewell) [inline pyproject.toml:26]: I think the psycop2g-binary pinning can be removed here now?

**Review by chrisjsewell — CHANGES_REQUESTED** [review]
All looks good cheers! Just a minor change request

File: pyproject.toml (hunk 1 above)
Comment (JPchico) [reply pyproject.toml:26]: You want me to remove the `psycopg2-binary<2.9` by `psycopg2-binary`? It is just that the diff in the `pyproject.toml` is showing a change in the version of aiida-core that already exists.

Comment (chrisjsewell) [reply pyproject.toml:26]: yep, this was indeed fixed upstream: https://github.com/aiidateam/aiida-core/pull/5104

Comment (JPchico) [reply pyproject.toml:26]: I guess we don't really to explicitly set `psycog-binary2 ` as a dependency right? as it is an `aiida-core` dependency. The same for `ase` as it is installed via `aiida-core[atomic-tools]`.

**Review by chrisjsewell — APPROVED** [review]
All sorted!

**Pull request merged**